### PR TITLE
lightning: evaluate all generated columns even if they are virtual

### LINKED ISSUE #1404: [lightning] admin check failed after importing table with expression index using local backend
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

- Create table t3 with expression index

    ```sql
      CREATE TABLE `t3` (
        `id` int(11) NOT NULL AUTO_INCREMENT,
        `a` varchar(20) DEFAULT NULL,
        `b` varchar(20) DEFAULT NULL,
        PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */,
        KEY `idx_a` (`a`),
        KEY `idx_lower_b` ((lower(`b`)))
      ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin AUTO_INCREMENT=90003;
    ```

- Use dumpling to export the table
- rename table t3 to t3bak
- Use lightning local backend to import the table to TiDB

2. What did you expect to see?
Lightning import should be successful, and admin check for the table should pass after that.

3. What did you see instead?
Lightning import is successful, however admin check failed.

```
MySQL [test]> admin check table t3;
ERROR 8134 (HY000): col _V$_idx_lower_b_0, handle 1, index:types.Datum{k:0x5, decimal:0x0, length:0x0, i:0, collation:"utf8mb4_bin", b:[]uint8{0x1}, x:interface {}(nil)} != record:types.Datum{k:0x5, decimal:0x0, length:0x0, i:0, collation:"utf8mb4_bin", b:[]uint8{0x62, 0x62, 0x62}, x:interface {}(nil)}, compare err:<nil>

MySQL [test]> admin check index t3 idx_lower_b;
ERROR 8134 (HY000): col _V$_idx_lower_b_0, handle 1, index:types.Datum{k:0x5, decimal:0x0, length:0x0, i:0, collation:"utf8mb4_bin", b:[]uint8{0x1}, x:interface {}(nil)} != record:types.Datum{k:0x5, decimal:0x0, length:0x0, i:0, collation:"utf8mb4_bin", b:[]uint8{0x62, 0x62, 0x62}, x:interface {}(nil)}, compare err:<nil>

MySQL [test]> admin check table t3bak;
Query OK, 0 rows affected (0.01 sec)
```

4. What version of BR and TiDB/TiKV/PD are you using?
dumpling - v5.1.1
tidb-lightning -V5.1.1
tidb-server -V5.1.1
tikv-server -V5.1.1

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

Fix #1404 

### What is changed and how it works?

Expression indices rely on a virtual generated column, which is not calculated by Lightning (unless depended by a stored generated column). This caused the virtual generated column to contain invalid value and thus invalidate the index.

This PR just calculates all generated columns no matter they are virtual or stored.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

 - Possible performance regression
    * Tables with only virtual generated columns will experience longer encoding time.

Related changes

 - Need to cherry-pick to the release branch
    * release-4.0, 5.0, 5.1 (everywhere that creating an index on virtual generated column is allowed)

### Release note

 - Expression index and index depending on virtual generated columns are now valid. Previously these indices are broken when importing through Lightning local or importer backend.

### CODE REVIEW

**Comment by glorv** [comment]
Why do this PR need to be cherry-picked into 4.0 and 5.0?

**Comment by kennytm** [comment]
@glorv the following schema works on 4.0 and I believe Lightning before this PR won't populate the `b` unique index correctly.

```sql
create table a(a int primary key, b int as (a + 1) virtual unique);
```

**Review by glorv — APPROVED** [review]

File: tests/config/tidb.toml [hunk 1]
```diff
@@ -11,3 +11,6 @@ ssl-key = "/tmp/backup_restore_test/certs/tidb.key"
 cluster-ssl-ca = "/tmp/backup_restore_test/certs/ca.pem"
 cluster-ssl-cert = "/tmp/backup_restore_test/certs/tidb.pem"
 cluster-ssl-key = "/tmp/backup_restore_test/certs/tidb.key"
+
+[experimental]
+allow-expression-index = true
```
Comment (wjhuang2016) [inline tests/config/tidb.toml:16]: It's removed.

Comment (kennytm) [reply tests/config/tidb.toml]: still needed when cherry-picking.

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**